kars/0005-hardware-topology-awareness/README.md

```diff
@@ -0,0 +1,44 @@
+# KAR-0005: Hardware Topology Awareness
+
+## Description
+
+Hardware topology awareness requires the Kubernetes cluster to understand the physical layout of specialized hardware, such as GPU-GPU interconnects, GPU-NIC proximity, and more, so the scheduler can place workloads on nodes where accelerators have optimal latency and bandwidth. This ensures high-performance AI/ML jobs are scheduled on hardware configurations that maximize throughput and avoid suboptimal paths. This information should be exposed via DRA attributes if supported by accelerator and driver to enable topology-aware scheduling.
```

Comment:
IMO this requirement should go further than to require only that topology be represented in DRA attributes somehow. Requiring only that still leaves users to do a bunch of digging to find out in which ResourceSlices those attributes are recorded, which attributes are relevant, and what they mean, which could all be completely different for each platform.

That probably gets into what @kannon92 and @aojea were discussing in #5.